tensorflow/core/kernels/decode_bmp_op.cc

```diff
@@ -67,7 +67,7 @@ class DecodeBmpOp : public OpKernel {
     // Decode image, allocating tensor once the image size is known
     Tensor* output = nullptr;
     OP_REQUIRES_OK(context, context->allocate_output(
-        0, TensorShape({abs(height), width, 3}), &output));
+        0, TensorShape({abs(height), width, channels_}), &output));
```

Comment:
This won't work if channels_ is 0.

I think my suggestion would be as follows:

If the user provides channels_ != 0, validate that channels_ = bpp / 8 as you do above.

But in the rest of the code, always use bpp / 8 as the channels you use to allocate the tensor and pass into the Decode function.  Does that make sense?